lib/message_encryptor.rb

```diff
@@ -0,0 +1,35 @@
+class MessageEncryptor
+  def initialize(data)
+    @data = data
+  end
+
+  def encrypt
+    data.is_a?(Array) ? encrypt_array : crypt.encrypt_and_sign(data)
+  end
+
+  def decrypt
+    data.is_a?(Array) ? decrypt_array : crypt.decrypt_and_verify(data)
+  end
+
+  private
+
+  attr_reader :data
+
+  def crypt
+    @crypt ||= ActiveSupport::MessageEncryptor.new secret
+  end
+
+  def secret
+    salt = Rails.application.secrets.secret_key_base
```

Comment:
It would indeed, but with limited impact as only the Sign in audit encypts/decrypts the array.

We can add an new task to specifically set a secret used for this purpose. I did not want to introduce a new terraform variable as part of this PR.